scipy/stats/_multivariate.py

```diff
@@ -2642,3 +2644,143 @@ def entropy(self):
     method_frozen.__doc__ = doccer.docformat(
         method.__doc__, wishart_docdict_noparams)
     method.__doc__ = doccer.docformat(method.__doc__, wishart_docdict_params)
+
+class special_ortho_group_gen(multi_rv_generic):
+    r"""
+    A matrix-valued SO(N) random variable.
+
+    The `dim` keyword specifies the dimension N.
+
+    Methods
+    -------
+    ``rvs(dim=None, size=1, random_state=None)``
+        Draw random samples from SO(N).
+
+    Parameters
+    ----------
+    dim : scalar
+        Dimension of matrices
+
+    TODO: frozen object not implemented yet.
+
+    Notes
+    -----
+    Return a random rotation matrix, drawn from the Haar distribution
+    (the only uniform distribution on SO(n)).
+
+    Examples
+    --------
+    >>> from scipy.stats import special_ortho_group
+    >>> x = special_ortho_group.rvs(5)
+
+    This generates one random matrix from SO(5).
+
+    """
+
+    def __init__(self, seed=None):
+        super(special_ortho_group_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__)
+
+    def __call__(self, dim=None, seed=None):
+        """
+        Create a frozen SO(N) distribution.
+
+        See `special_ortho_group_frozen` for more information.
+
+        """
+        return special_ortho_group_frozen(dim, seed=seed)
+
+    def _process_parameters(self, dim):
+        """
+        Dimension N must be specified; it cannot be inferred.
+        """
+
+        if dim is None or not np.isscalar(dim) or dim <= 1:
+            raise ValueError("Dimension of rotation must be specified, and must be a scalar greater than 1.")
+
+        return dim
+
+    def rvs(self, dim, size=1, random_state=None):
+        """
+        Draw random samples from SO(N).
+
+        Parameters
+        ----------
+        dim : integer
+            Dimension of rotation space (N).
+        size : integer, optional
+            Number of samples to draw (default 1).
+
+        Returns
+        -------
+        rvs : ndarray or scalar
+            Random size N-dimensional matrices, dimension (size, dim, dim)
+
+        Notes
+        -----
+        Return a random rotation matrix, drawn from the Haar distribution
+        (the only uniform distribution on SO(n)).
+        The algorithm is described in the paper
+        Stewart, G.W., "The efficient generation of random orthogonal
+        matrices with an application to condition estimators", SIAM Journal
+        on Numerical Analysis, 17(3), pp. 403-409, 1980.
+        For more information see
+        http://en.wikipedia.org/wiki/Orthogonal_matrix#Randomization
```

Comment:
OK, I have moved it. Note that this class is just wrapping the random_rot function of the MDP Toolkit. It is also present in buggy form somewhere else in scipy and should probably be removed from there, maybe with a note pointing to the new home.